src/common/commonutils/UserUtils.c

```diff
@@ -3140,134 +3000,64 @@ int CheckUserAccountsNotFound(const char* names, char** reason, OsConfigLogHandl
     return status;
 }
 
-int RemoveUserAccounts(const char* names, bool removeHomeDirs, OsConfigLogHandle log)
+int RemoveUserAccounts(const char* names, OsConfigLogHandle log)
 {
-    const char* userTemplate = "%s:";
-    size_t namesLength = 0;
-    char* name = NULL;
-    char* decoratedName = NULL;
-    SimplifiedUser* userList = NULL;
-    unsigned int userListSize = 0, i = 0, j = 0;
-    bool userAccountsNotFound = false;
     int status = 0, _status = 0;
+    struct passwd* pw;
 
     if (NULL == names)
     {
         OsConfigLogError(log, "RemoveUserAccounts: invalid argument");
         return EINVAL;
     }
 
-    if (0 == (status = CheckUserAccountsNotFound(names, NULL, log)))
-    {
-        OsConfigLogInfo(log, "RemoveUserAccounts: user accounts '%s' are not found in the users database", names);
-        userAccountsNotFound = true;
-    }
-    else if (EEXIST != status)
+    setpwent();
+    for (errno = 0, pw = getpwent(); pw != NULL; errno = 0, pw = getpwent())
     {
-        OsConfigLogInfo(log, "RemoveUserAccounts: CheckUserAccountsNotFound('%s') returned %d", names, status);
-        return status;
-    }
-
-    namesLength = strlen(names);
+        char* usernames = NULL;
+        char* token = NULL;
```

Comment:
Move up for consistence with rest of the code in this file (we discussed to do this but let's also keep consistence)